In [1]:
# Importing necessary libraries

In [2]:
import pandas as pd

In [7]:
purchase_data = pd.read_csv(r'C:\Users\Dell Latitude 3460\Downloads\purchase_behaviour.csv')

In [16]:
purchase_data.head()

LYLTY_CARD_NBR               LIFESTAGE PREMIUM_CUSTOMER
0            1000   YOUNG SINGLES/COUPLES          Premium
1            1002   YOUNG SINGLES/COUPLES       Mainstream
2            1003          YOUNG FAMILIES           Budget
3            1004   OLDER SINGLES/COUPLES       Mainstream
4            1005  MIDAGE SINGLES/COUPLES       Mainstream

In [8]:
transaction_data = pd.read_csv(r'C:\Users\Dell Latitude 3460\Downloads\transaction_data.csv')

In [17]:
transaction_data.head()

DATE  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0  43390          1            1000       1         5   
1  43599          1            1307     348        66   
2  43605          1            1343     383        61   
3  43329          2            2373     974        69   
4  43330          2            2426    1038       108   

                                  PROD_NAME  PROD_QTY  TOT_SALES  
0    Natural Chip        Compny SeaSalt175g         2        6.0  
1                  CCs Nacho Cheese    175g         3        6.3  
2    Smiths Crinkle Cut  Chips Chicken 170g         2        2.9  
3    Smiths Chip Thinly  S/Cream&Onion 175g         5       15.0  
4  Kettle Tortilla ChpsHny&Jlpno Chili 150g         3       13.8

In [9]:
purchase_data.shape

(72637, 3)

In [10]:
transaction_data.shape

(264836, 8)

### As we can see The Purchase behavior data contains 72637 instances and 3 features
### whereas The Transaction data contains 264836 instances and 8 features

In [11]:
# let's look for the missing values

In [12]:
purchase_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72637 entries, 0 to 72636
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   LYLTY_CARD_NBR    72637 non-null  int64 
 1   LIFESTAGE         72637 non-null  object
 2   PREMIUM_CUSTOMER  72637 non-null  object
dtypes: int64(1), object(2)
memory usage: 1.7+ MB


In [13]:
transaction_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264836 entries, 0 to 264835
Data columns (total 8 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   DATE            264836 non-null  int64  
 1   STORE_NBR       264836 non-null  int64  
 2   LYLTY_CARD_NBR  264836 non-null  int64  
 3   TXN_ID          264836 non-null  int64  
 4   PROD_NBR        264836 non-null  int64  
 5   PROD_NAME       264836 non-null  object 
 6   PROD_QTY        264836 non-null  int64  
 7   TOT_SALES       264836 non-null  float64
dtypes: float64(1), int64(6), object(1)
memory usage: 16.2+ MB


In [18]:
top_purchased_products = (transaction_data.groupby(["PROD_NBR", "PROD_NAME"])["TOT_SALES"].sum().reset_index().sort_values(by="TOT_SALES", ascending=False).head(3))

In [19]:
loyal_customers = (transaction_data.groupby("LYLTY_CARD_NBR")["TXN_ID"].count().reset_index().rename(columns={"TXN_ID": "TXN_COUNT"}).sort_values(by="TXN_COUNT", ascending=False))

In [20]:
loyal_customers = loyal_customers.merge(purchase_data, on="LYLTY_CARD_NBR")

In [21]:
loyal_customer_profile = loyal_customers.groupby(["LIFESTAGE", "PREMIUM_CUSTOMER"])\
    .size().reset_index(name="COUNT")\
    .sort_values(by="COUNT", ascending=False)\
    .head(3)

In [23]:
top_purchased_products

PROD_NBR                                 PROD_NAME  TOT_SALES
3          4          Dorito Corn Chp     Supreme 380g    40352.0
13        14    Smiths Crnkle Chip  Orgnl Big Bag 380g    36367.6
15        16  Smiths Crinkle Chips Salt & Vinegar 330g    34804.2

In [24]:
loyal_customer_profile

LIFESTAGE PREMIUM_CUSTOMER  COUNT
19  YOUNG SINGLES/COUPLES       Mainstream   8088
13               RETIREES       Mainstream   6479
10  OLDER SINGLES/COUPLES       Mainstream   4930